src/commonComponents/sidebar/Sidebar.js

```diff
@@ -0,0 +1,92 @@
+// react
+import { useEffect, useState } from "react";
+
+// prop-types
+import PropTypes from "prop-types";
+
+//lodash
+import _map from "lodash/map";
+import _startCase from "lodash/startCase";
+import _findIndex from "lodash/findIndex";
+
+// constants
+import { INITIAL_INDEX, SCROLL_FACTOR } from "./constants/sidebar.general";
+
+//helpers
+import { sanitizeText } from "../../utility/sanitizeText";
+import {
+  getMatchedIndexWithUrlHash,
+  isEqualWith,
+} from "./helpers/sidebar.general";
+
+// components
+import LinkItem from "../linkItem";
+
+// css
+import "./sidebar.css";
+
+const Sidebar = (props) => {
+  const { links, className } = props;
+
+  const [activeIdx, setActiveIdx] = useState(INITIAL_INDEX);
+
+  useEffect(() => {
+    const setInitialActiveIdx = () => {
```

Comment:
this function is not needed, you can directly write those lines in effect function, it's a redundant function created when everytime the effect is called